### PR TITLE
add generic LED utility. Add usb-light reference design.

### PR DESCRIPTION
**Pull request by electricfungi**
Added gen-led-cmp() and usb-light.stanza

### CODE REVIEW

File: designs/usb-light.stanza [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  inst led : gen-led-cmp(tol(2.1,0.1), 0.02, "YELLOW", "0805")[8]
+  net p5v (vbus.vdd conn.VCC)
+  net gnd (vbus.gnd conn.GND conn.SHIELD)
+  symbol(p5v) = ocdb/symbols/supply-sym
+  symbol(gnd) = ocdb/symbols/ground-sym
+  val vbus-voltage = (nominal(recommended-voltage(property(conn.VCC.power-pin)))) as Double
+  val resistor-voltage = vbus-voltage - (2.0 * property(led[0].forward-voltage))
+  println("Vbus = %_" % [vbus-voltage])
+  println("Vf = %_" % [property(led[0].forward-voltage)])
+  val exact-resistance = resistor-voltage / property(led[0].test-current)
```
Comment (PhilippeFerreiraDeSousa) [inline designs/usb-light.stanza:30]: You should probably isolate the maths in a function

File: utils/generic-components.stanza [hunk 2]
```diff
@@ -721,6 +723,66 @@ public defn gen-ind-cmp (ind:Double, tol:Double) :
...
   gen-ind-cmp(ind, 10.0)
+; ========================================================
+; ===================== gen-led-cmp ======================
+; ========================================================
+public pcb-component gen-led-cmp (vf:Toleranced, itest:Double, color:String, pkg-name:String) :
+  pin a
+  pin c
+  description = string-join(["LED, GENERIC, ", color, " Vf ", to-string(center(vf)), "V, If ", to-string(to-int(itest * 1000.0)), "mA, ", pkg-name])
+  mpn = string-join([color ", Vf ", to-string(center(vf)), "V, If ", to-string(to-int(itest * 1000.0)), "mA"])
+  name = string-join(["Generic", color, " LED ", pkg-name])
```
Comment (PhilippeFerreiraDeSousa) [inline utils/generic-components.stanza:734]: Hard to read. What about?
```
  val mpn = to-string("%_, Vf %_V, If %_mA" % [color, center(vf),  to-int(itest * 1000.0)])
  description = to-string(["LED, GENERIC, %_, %_" % [mpn, pkg-name])
  mpn = mpn
  name = to-string("Generic%_ LED %_" % [color, pkg-name])
```

File: designs/usb-light.stanza (hunk 1 above)
Comment (electricfungi) [reply designs/usb-light.stanza]: created function led-circuit-resistance (

File: utils/generic-components.stanza (hunk 2 above)
Comment (electricfungi) [reply utils/generic-components.stanza]: switched to to-string

**Review by PhilippeFerreiraDeSousa — APPROVED** [review]

**Pull request merged**